tests/nixseparatedebuginfod/devenv.nix

```diff
@@ -0,0 +1,28 @@
+{ lib, pkgs, config, ... }:
+
+let
+  cfg = config.services.nixseparatedebuginfod;
+  inherit (pkgs.stdenv) hostPlatform;
+
+  # Use a locally built derivation so that the test wouldn't rely on cache.nixos.org
+  cbin = pkgs.stdenv.mkDerivation {
+    pname = "cbin";
+    version = "1.0";
+    src = ./.;
+    installFlags = [ "DESTDIR=$(out)" ];
+    separateDebugInfo = true;
+    meta.mainProgram = "example";
+  };
+in
+lib.mkIf (lib.meta.availableOn hostPlatform cfg.package) {
+  services.nixseparatedebuginfod.enable = true;
+
+  # The Nix store needs to be indexed by nixseparatedebuginfod for debug outputs from local
+  # derivations to be served. This can take a few minutes.
+  process.before = "${lib.getExe cfg.package} --index-only";
```

Comment:
I added a test that fetches debuginfo from the server, but I'm not sure if this is a good idea. nixseparatedebuginfod needs to index the Nix store before it can serve debuginfos for locally built derivations. This can "take a few minutes" according to the official README. On my machine, it took a little more than a minute so I set the timeout to 2 minutes. The actual time will likely depend on the size of the Nix store and I don't know if 2 minutes is going to be enough, though.